### PR TITLE
new(ButtonGroup): Add `stretched` button prop.

### PR DESCRIPTION
**Pull request by stefhatcher**
to: @milesj @stefhatcher @hayes @alecklandgraf

## Description

stretch buttons in a ButtonGroup

## Motivation and Context

sometimes, need those button to fill the space

## Testing

local storybook
happo

## Screenshots

Core : ButtonGroup - Stretch Buttons ⋅ Storybook 2019-11-21 15-00-42

## Checklist

- My code follows the style guide of this project.
- I have updated or added documentation accordingly.
- I have read the CONTRIBUTING document.

### CODE REVIEW

File: packages/core/src/components/ButtonGroup/index.tsx [hunk 1]
```diff
@@ -25,7 +28,17 @@ export default function ButtonGroup({ children, endAlign, stacked }: Props) {
...
-          <div className={cx(stacked ? styles.cell_stacked : styles.cell)}>{child}</div>
+          <div
+            className={cx(
+              styles.cell,
+              stacked && styles.cell_stacked,
+              stretched && styles.cell_block,
+            )}
+          >
+            {React.cloneElement(child as React.ReactElement<ButtonOrLinkProps>, {
+              block: stretched,
```
Comment (milesj) [inline packages/core/src/components/ButtonGroup/index.tsx]: Can we not force block via flexbox?

Comment (stefhatcher) [reply packages/core/src/components/ButtonGroup/index.tsx]: we could if i targeted `button` or `a` or `div` or `span` direct child of the cell. happy to go that route too, if you want

Comment (milesj) [reply packages/core/src/components/ButtonGroup/index.tsx]: Worth a shot. I'm not a fan of cloning elements.

Comment (stefhatcher) [reply packages/core/src/components/ButtonGroup/index.tsx]: sounds good !

**Comment by stefhatcher** [comment]
updated @milesj

File: packages/core/src/components/ButtonGroup/styles.ts [hunk 2]
```diff
@@ -28,12 +28,23 @@ const styleSheet: StyleSheet = ({ unit }) => ({
...
     ':last-of-type': {
       marginBottom: 0,
     },
   },
+
+  cell_stretched: {
+    flex: 1,
+
+    '@selectors': {
+      '> button, > a, > div, > span': {
```
Comment (milesj) [inline packages/core/src/components/ButtonGroup/styles.ts]: `> * { flexGrow: 1 }`?

Comment (stefhatcher) [reply packages/core/src/components/ButtonGroup/styles.ts]: tried *, will try again

Comment (stefhatcher) [reply packages/core/src/components/ButtonGroup/styles.ts]: flexGrow doesn't work because they're grandchildren of the flex

**Pull request merged**